### PR TITLE
Adding PAI Slot to PDA

### PR DESCRIPTION
**Pull request by evilexecutive**
## About the PR
I have added an additional slot to all PDAs that will accept PAI cartridges. This has been a feature in SS13 for a very long time, so I'm genuinely surprised it's not a thing you can do in SS14. I made this as a side project for doing research on how to impliment security body/helmet cams, and other features that require attaching an item to another item. 

## Technical details
This touches the files PdaComponent.cs and SharedPdaSystem.cs, as well as yaml files for base PDA and the pai card.

## Media
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase
image (2)

**Changelog**
:cl: VMSolidus
- add: PAI cartridges can now be slotted into PDAs.

### CODE REVIEW

File: Resources/Prototypes/tags.yml [hunk 1]
```diff
@@ -1082,3 +1082,6 @@
 
 - type: Tag
   id: MindShield
+
+- type: Tag
+  id: PAISlot
```
Comment (lzk228) [inline Resources/Prototypes/tags.yml:1087]: tags should be sorted

Comment (VMSolidus) [reply Resources/Prototypes/tags.yml]: Got it!

**Review by ficcialfaint — DISMISSED** [review]
I think that should also have a UI button to eject the PAI as pen or ID slots do

**Comment by VMSolidus** [comment]
> I think that should also have a UI button to eject the PAI as pen or ID slots do

I would like to apologize for this, as I do not currently possess the coding skills required to create UI elements. It would not be possible for me to implement your change request at this time.

**Comment by VMSolidus** [comment]
Okay after consulting with some other programmers, I'm going to give it a try with making the requested UI element.

**Comment by VMSolidus** [comment]
image
Yep, got the PDA-PAI UI elements working.

File: Resources/Prototypes/Entities/Objects/Devices/pda.yml [hunk 2]
```diff
@@ -23,6 +23,11 @@
     state: pda
   - type: Pda
     state: pda
+    pAISlot:
+      priority: -2
+      whitelist:
+        tags:
+        - PAISlot
```
Comment (ficcialfaint) [inline Resources/Prototypes/Entities/Objects/Devices/pda.yml:30]: perhaps whitelisting entities with PAI component?
```suggestion
        components:
        - PAI
```

**Review by ficcialfaint — DISMISSED** [review]

**Comment by LankLTE** [comment]
@evilexecutive Please say if you intend to come back to this.

**Comment by VMSolidus** [comment]
@LankLTE Hey I straight up didn't know that this PR even had a review on it. I was under the impression it was done, and now looking back on it, am confused that the review essentially only wanted me to change some letter casing. I can do that right now if you reopen the PR for me.

**Comment by DogZeroX** [comment]
image
:trollface:

File: Content.Client/PDA/PdaMenu.xaml.cs [hunk 3]
```diff
@@ -41,6 +41,7 @@ public PdaMenu()
             FlashLightToggleButton.IconTexture = new SpriteSpecifier.Texture(new("/Textures/Interface/light.png"));
             EjectPenButton.IconTexture = new SpriteSpecifier.Texture(new("/Textures/Interface/pencil.png"));
             EjectIdButton.IconTexture = new SpriteSpecifier.Texture(new("/Textures/Interface/eject.png"));
+            EjectPaiButton.IconTexture = new SpriteSpecifier.Texture(new("/Textures/Interface/PAI.png"));
```
Comment (ficcialfaint) [inline Content.Client/PDA/PdaMenu.xaml.cs:44]:
```suggestion
            EjectPaiButton.IconTexture = new SpriteSpecifier.Texture(new("/Textures/Interface/pai.png"));
```

**Review by ficcialfaint — COMMENTED** [review]
:rage1:

**Pull request merged**